modules/apps/object/object-rest-api/src/main/java/com/liferay/object/rest/manager/v1_0/BaseObjectEntryManager.java

```diff
@@ -126,9 +126,8 @@ protected long getGroupId(
 		if (objectScopeProvider.isGroupAware()) {
 			if (Objects.equals(objectDefinition.getScope(), "site")) {
 				return GetterUtil.getLong(
-					GroupUtil.getGroupId(
-						objectDefinition.getCompanyId(), scopeKey,
-						groupLocalService));
+					objectScopeProvider.getValidGroupId(
+						objectDefinition.getCompanyId(), scopeKey));
```

Comment:
@devpedrotavares can we try adapting the code from `GroupUtil.getGroupId` instead of duplicating it? We can validate the change with the headless team.

Reply:
It is possible to do the validation there. But perhaps this is an opportunity to decouple our logic from theirs, since they are validating according to their requirements and we might need to adapt again in the future. 
Please, let me know what we should do.

Reply:
@devpedrotavares the problem I see it's that the `GroupUtil.getGroupId` is being used in the [VulcanBatchEngineTaskItemDelegateAdaptor](https://github.com/liferay/liferay-portal/blob/05a6f46bc06c965a97368d57502993db0d78c62d/modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/batch/engine/VulcanBatchEngineTaskItemDelegateAdaptor.java#L192-L205) class, and in other places that may be used by us. 

Could you check this, as we must keep consistency?

Reply:
Sure, I've opened a thread with the headless team [here](https://liferay.slack.com/archives/C5E1CRLJY/p1717703609889769?thread_ts=1717703495.334109&cid=C5E1CRLJY)

Reply:
Hi @carolmariaabb , they replied in the thread saying that adding this change in `GroupUtil.getGroupId` would introduce a new behavior for all REST APIs that contains a `siteId` as a path parameter by allowing User Group sites in all of them.